spec.html

```diff
@@ -28857,10 +28857,18 @@ <h1>
         </dl>
         <emu-alg>
           1. Assert: If _direct_ is *false*, then _strictCaller_ is also *false*.
-          1. If _x_ is not a String, return _x_.
+          1. If _x_ is an Object, then
+            1. Let _code_ be HostGetCodeForEval(_x_).
+            1. If _code_ is ~unknown~ return _x_.
+            1. Else, let _xStr_ be _code_.
+          1. Else if _x_ is a String, then
+            1. Let _xStr_ be _x_.
+          1. Else,
+            1. Return _x_.
           1. Let _evalRealm_ be the current Realm Record.
           1. NOTE: In the case of a direct eval, _evalRealm_ is the realm of both the caller of `eval` and of the `eval` function itself.
-          1. Perform ? HostEnsureCanCompileStrings(_evalRealm_, « », _x_, _direct_).
+          1. Let _type_ be ~direct-eval~ if _direct_ is *true*, or ~indirect-eval~ if _direct_ is *false*.
```

Comment:
```suggestion
          1. If _direct_ is *true*, let _type_ be ~direct-eval~. Otherwise, let _type_ be ~indirect-eval~.
```

Reply:
How do you feel about changing the type of `PerformEval`'s `_direct_` parameter to be `~direct-eval~ | ~indirect-eval~` instead? So that:
- we don't have to convert from boolean to enum
- it's clearer what the caller of `PerformEval` mean then when using "true"/"false"

Reply:
This is what I was suggesting @nicolo-ribaudo https://github.com/tc39/ecma262/pull/3294#discussion_r1554364846

Reply:
Changed this